examples/c/profile.c

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+/* Copyright (c) 2022 Facebook */
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <sys/sysinfo.h>
+#include <linux/perf_event.h>
+#include <bpf/libbpf.h>
+#include <bpf/bpf.h>
+
+#include "profile.skel.h"
+#include "profile.h"
+#include "blazesym.h"
+
+static long perf_event_open(struct perf_event_attr *hw_event, pid_t pid,
+		int cpu, int group_fd, unsigned long flags)
+{
+	int ret;
+
+	ret = syscall(__NR_perf_event_open, hw_event, pid, cpu,
+		      group_fd, flags);
+	return ret;
+}
+
+typedef enum {
+	KERNEL,
+	USER
+} sym_index_t;
+
+static blazesym *symbolizer;
+
+static void show_stack_trace(uint64_t *stack, int stack_sz, pid_t pid)
+{
+	const struct blazesym_result *syms;
+	sym_file_cfg cfg;
+	int i;
+
+	if (pid) {
+		cfg.cfg_type = CFG_T_PROCESS;
+		cfg.params.process.pid = pid;
+	} else {
+		cfg.cfg_type = CFG_T_KERNEL;
+		cfg.params.kernel.kallsyms = NULL;
+		cfg.params.kernel.kernel_image = NULL;
+	}
+
+	syms = blazesym_symbolize(symbolizer, &cfg, 1, (const uint64_t *)stack, stack_sz);
+	for (i = 0; i < stack_sz; i++) {
+		if (syms && syms[i].valid) {
+			if (syms[i].path && syms[i].path[0]) {
+				printf("  %d [<%016lx>] %s+0x%lx %s:%ld\n",
+				       i, stack[i], syms[i].symbol,
+				       stack[i] - syms[i].start_address,
+				       syms[i].path, syms[i].line_no);
+			} else {
+				printf("  %d [<%016lx>] %s+0x%lx\n",
+				       i, stack[i], syms[i].symbol,
+				       stack[i] - syms[i].start_address);
+			}
+		} else
+			printf("  %d [<%016lx>]\n", i, stack[i]);
```

Comment:
if one part of if/else has {} then the other has to as well (even if it's a simple single-line statement)